pdns/dnsrecords.cc

```diff
@@ -45,15 +45,26 @@ void DNSResourceRecord::setContent(const string &cont) {
 
 string DNSResourceRecord::getZoneRepresentation(bool noDot) const {
   ostringstream ret;
+  vector<string> parts;
+  string last;
+
   switch(qtype.getCode()) {
     case QType::SRV:
     case QType::MX:
+      stringtok(parts, content);
+      last = *parts.rbegin();
+      ret << content;
+      if (last == ".")
```

Comment:
Just a nit but this test looks redundant with the next one.

Reply:
As discussed on IRC, leaving this explicitness in.